akka-http/src/main/scala/akka/http/scaladsl/server/PathMatcher.scala

```diff
@@ -358,6 +358,22 @@ trait PathMatchers {
 
   /**
    * A PathMatcher that matches and extracts the complete remaining,
+   * unmatched part of the request's URI path as an decoded (with actual whitespace) String.
+   * @group pathmatcher
+   */
+  object RemainingDecoded extends PathMatcher1[String] {
+    private val encodedSpace = "%20"
+    private val decodedSpace = " "
+
+    def decodeWhitespace(path: String) = path.replace(encodedSpace, decodedSpace)
```

Comment:
This only decodes the spaces (which were just encoded by `path.toString` below), but leaves other characters escaped, which could be surprising.

Using `RemainingPath` as described in the `Remaining` doc above and simply concatenating the segments would be a better approach IMO.